gem5-configs/configs-gapbs-tests/gapbs_config.py

```diff
@@ -0,0 +1,122 @@
+#Copyright (c) 2020 The Regents of the University of California.
+#All Rights Reserved
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+
+""" Script to run GAP Benchmark suites workloads. 
+    The workloads have two modes: synthetic and real graphs.
+"""
+
+import sys
+import time
+
+import m5
+import m5.ticks
+from m5.objects import *
+
+sys.path.append('gem5/configs/common/') # For the next line...
+import SimpleOpts
+
+from system import *
+
+SimpleOpts.set_usage(
+    "usage: %prog [options] kernel disk cpu_type num_cpus mem_sys benchmark synthetic graph")
+
+SimpleOpts.add_option("--allow_listeners", default=False, action="store_true",
+                      help="Listeners disabled by default")
+
+
+def writeBenchScript(dir, benchmark_name, size, synthetic):
```

Comment:
This doesn't seem to match with the `runscript.sh` above. In runscript, the data you're passing into the guest is the workload, arg, and size. Here, you seem to be creating a script. Maybe I'm missing something, but this seems like an inconsistency.

Reply:
I might be wrong but this is what I wanted to do:
writeBenchScript passes the workload, arg, and size into an script and the runscript reads that script from the host into the workloadfile.